ecs-cli/modules/cli/compose/entity/service/service_helper.go

```diff
@@ -15,17 +15,74 @@ package service
 
 import (
 	"fmt"
+	"sort"
+	"time"
 
 	log "github.com/Sirupsen/logrus"
+	ecscli "github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
 	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/waiters"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
 )
 
+// TimeOutUpdateService is the time that the CLI will wait to check if the
+// count of running tasks is changing. If it has not changed then an error is thrown
+// after TimeOutUpdateService minutes
+const TimeOutUpdateService = 5
+const PrintOnlyNewEvents = 10
+
+// serviceEvents is a wrapper for []*ecs.ServiceEvent
+// that allows us to reverse it
+type eventSorter []*ecs.ServiceEvent
+
+func (s eventSorter) Len() int {
+	return len(s)
+}
+func (s eventSorter) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+func (s eventSorter) Less(i, j int) bool {
+	time1 := *s[i].CreatedAt
+	time2 := *s[j].CreatedAt
+	diff := time1.Sub(time2)
+	return diff.Seconds() < 0
+}
+
+// logNewServiceEvents logs events that have not been logged yet
+func logNewServiceEvents(loggedEvents map[string]bool, events []*ecs.ServiceEvent, commandTime time.Time) {
+
+	// the slice comes ordered so that newer events are first. Logically, we
+	// want to print older events first- so we reverse it
+	sort.Sort(eventSorter(events))
+	for _, event := range events {
+		if _, ok := loggedEvents[*event.Id]; !ok {
+			// New event that has not been logged yet
+			loggedEvents[*event.Id] = true
+			if commandTime.Sub(*event.CreatedAt).Seconds() < PrintOnlyNewEvents {
+				log.WithFields(log.Fields{
+					"timestamp": *event.CreatedAt},
+				).Info(*event.Message)
+			}
+		}
+	}
+
+}
+
 // waitForServiceTasks continuously polls ECS (by calling describeService) and waits for service to get stable
 // with desiredCount == runningCount
 func waitForServiceTasks(service *Service, ecsServiceName string) error {
 	timeoutMessage := fmt.Sprintf("Timeout waiting for service %s to get stable", ecsServiceName)
 
+	eventsLogged := make(map[string]bool)
+	var lastRunningCount int64
+	lastRunningCountChangedAt := time.Now()
+	timeOut := float64(TimeOutUpdateService)
+	timeWhenFunctionCalled := time.Now()
+
+	if val := service.Context().CLIContext.Float64(ecscli.ComposeServiceTimeOutFlag); val > 0 {
```

Comment:
The reason that I asked for it initially is that was one of the temporary solutions for one of our customer to prevent their script from failing because of timeout errors. They just wanted to make sure the Service API calls succeeded and considered the script has successfully finished its job.

With the new implementation and configurable timeouts maybe this is no longer a use case (zero should not be allowed in this case). What do you think?